src/wesnothd_connection.hpp

```diff
@@ -147,8 +149,15 @@ class wesnothd_connection
 	typedef boost::asio::ip::tcp::resolver resolver;
 	resolver resolver_;
 
-	typedef boost::asio::ip::tcp::socket socket;
-	socket socket_;
+	boost::asio::ssl::context tls_context_;
+
+	std::string host_;
+	std::string service_;
+	typedef boost::asio::ip::tcp::socket raw_socket;
+	typedef boost::asio::ssl::stream<raw_socket> tls_socket;
+	typedef utils::variant<raw_socket, tls_socket> any_socket;
+	bool use_tls_ = true;
```

Comment:
I think if you use inline initialization you need to provide an inline initializer for all members. I would not be opposed to that, though.

Reply:
Though in this case it would probably just be easier to move this to the initializer list.